### ISSUE TITLE
scenario_autoretry does not reset runner.hook_failures

### ISSUE THREAD

**Issue by rittneje** [behave/behave#1128]
The scenario_autoretry monkey patch does not intervene with or reset the increment of `self.hook_failures` in the runner. Consequently, if before_scenario/after_scenario fails the first time, and then passes on the retry, behave still exits with failure at the end because `hook_failures` is non-zero.

https://github.com/behave/behave/blob/1c6197b35c15e07b5bae62b3131a98f9caa88f4e/behave/runner.py#L649
https://github.com/behave/behave/blob/1c6197b35c15e07b5bae62b3131a98f9caa88f4e/behave/runner.py#L751-L755

**Comment by jenisys** [comment]
Mmh, you seem to have flaky tests.
Therefore, best practice is to fix the flaky tests.

The `scenario_autoretry` is proof-of-concept code / best-effort code.
Come up with a better solution.

SOLUTION SKETCH:
You probably need to record the `test-runner state` before the scenario is run to be able to restore this state if you retry a scenario. If the `test-runner state` currently contains only the `hooks_failed` count, the solution will become even simpler.

SEE ALSO:

* [behave/contrib/scenario_autoretry.py](https://github.com/behave/behave/blob/main/behave/contrib/scenario_autoretry.py)

**Comment by rittneje** [comment]
@jenisys Yes obviously we'd the tests not to be flaky. But reality ensues.

I don't know what you mean by the "test-runner state".

**Comment by jenisys** [comment]
`Test-runner state` (in this case) is any group of variables in the `test-runner` that you need to restore when you retry a scenario.
